File: java-pi-pantry/src/test/java/com/langrsoft/pi/pantry/AnItemRetrieverWithSpecificApiPropertyKey.java
package com.langrsoft.pi.pantry;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import static org.hamcrest.Matchers.endsWith;
import static org.junit.Assert.assertThat;

public class AnItemRetrieverWithSpecificApiPropertyKey {
    String currentApiKey;

    @Before
    public void saveCurrentApiKey() {
        currentApiKey = System.getProperty(ItemRetriever.UPC_API_KEY_PROPERTY_NAME);
    }

    @After
    public void resetCurrentApiKeyToSaved() {
        if (currentApiKey != null)
            System.setProperty(ItemRetriever.UPC_API_KEY_PROPERTY_NAME, currentApiKey);
    }

    @Test
    public void constructsUrlUsingSystemPropertyForApiKey() {
        String someApiKey = "SOME_API_KEY";
        System.setProperty(ItemRetriever.UPC_API_KEY_PROPERTY_NAME, someApiKey);
        ItemRetriever retriever = new ItemRetriever(null);

        String url = retriever.url("123");

        assertThat(url, endsWith(someApiKey + "/123"));
    }
}

File: csharp-pi-pantry/csharp-pi-pantry-test/DomainTests/AnItemRetrieverWithSpecificApiPropertyKey.cs
using NUnit.Framework;
using Pipantry.Domain;
using System;

namespace Test.Pipantry.Domain
{
    [TestFixture]
    public class AnItemRetrieverWithSpecificApiPropertyKey
    {
        string currentApiKey;

        [SetUp]
        public void SaveCurrentApiKey()
        {
            currentApiKey = Environment.GetEnvironmentVariable(ItemRetriever.UPC_API_KEY_PROPERTY_NAME);
        }

        [TearDown]
        public void ResetCurrentApiKeyToSaved()
        {
            if (currentApiKey != null)
                Environment.SetEnvironmentVariable(ItemRetriever.UPC_API_KEY_PROPERTY_NAME, currentApiKey);
        }

        [Test]
        public void ConstructsUrlUsingSystemPropertyForApiKey()
        {
            Environment.SetEnvironmentVariable(ItemRetriever.UPC_API_KEY_PROPERTY_NAME, "SOME_API_KEY");
            var retriever = new ItemRetriever(null);

            var url = retriever.Url("123");

            Assert.That(url, Does.EndWith("SOME_API_KEY/123"));
        }
    }
}
